packages/x-charts/src/LineChart/formatter.ts

```diff
@@ -60,9 +59,8 @@ const formatter: SeriesFormatter<'line'> = (params, dataset) => {
           ? dataset!.map((data) => {
               const value = data[dataKey];
               if (typeof value !== 'number') {
-                if (process.env.NODE_ENV !== 'production' && !warnedOnce && value !== null) {
-                  warnedOnce = true;
-                  console.error([
+                if (process.env.NODE_ENV !== 'production' && value !== null) {
+                  warnOnce([
```

Comment:
Codebase convention. The idea was to make those ease to spot.

```suggestion
                if (process.env.NODE_ENV !== 'production') {
                  if (value !== null) {
                    warnOnce([
```

If we implement https://github.com/mui/mui-public/issues/207, we might not need this convention as much, not sure.